src/extension.ts

```diff
@@ -989,11 +991,22 @@ function execKubernetesCore(isTerminal) {
 }
 
 function execTerminalOnPod(podName : string, terminalCmd : string) {
-    const terminalExecCmd : string[] = ['exec', '-it', podName, terminalCmd];
+    const terminalExecCmd : string[] = ['exec', '-it', podName, '--', terminalCmd];
     const term = vscode.window.createTerminal(`${terminalCmd} on ${podName}`, kubectl.path(), terminalExecCmd);
     term.show();
 }
 
+function checkBash(podName : string): Promise<boolean> {
```

Comment:
Suggest changing this to `async` - I've been trying to run down the use of explicit promise management in favour of async/await because I feel it's a lot more readable.  To do this, you'll need to expose `kubectl.invokeAsync` on the interface (and I think you can/should remove the `ShellHandler` argument); I think it's worth it though.

Reply:
Nice suggestion. Use async/await is more clean. Thanks.